### ISSUE TITLE
Support Serde 1.0

### ISSUE THREAD

**Issue by drusellers** [SergioBenitez/Rocket#273]
Captain Obvious here:

Now that Serde 1.0 has been released. I'm excited to use all the libraries with it. I see that currently RocketContrib doesn't yet support 1.0 and a quick `cargo build` shows at least one breaking change around lifetime parameters.

I'm happy to try and take a whack at it, unless its already done and I'm just dense.

## Feature Requests

Feature requests _must_ include:

  1. Why you believe this feature is necessary.

The rust community has adopted Serde, and with 1.0 having been released I believe there will be many requests for Rocket to support a 1.0 version.

  2. A convincing use-case for this feature.

We already use Serde, so this is more of a maintenance task and should hopefully help Rocket Contrib achieve its own 1.0 status.

  3. Why this feature can't or shouldn't exist outside of Rocket.

This is improving existing code

**Comment by dtolnay** [comment]
This is currently blocked on https://github.com/sunng87/handlebars-rust/pull/151.

**Comment by drusellers** [comment]
The issue has been [merged](https://github.com/sunng87/handlebars-rust/pull/151) and it appears a crate released: https://crates.io/crates/handlebars

**Comment by drusellers** [comment]
Do we need to wait on https://crates.io/crates/rmp-serde to drop a `serde="1.0"` release as well?

**Comment by SergioBenitez** [comment]
@drusellers Yes, that's right.

**Comment by drusellers** [comment]
Cool. I'll reach out to them as well

**Comment by drusellers** [comment]
RMP has dropped - https://github.com/3Hren/msgpack-rust/issues/125#issuecomment-296815067
Handlebars is good - https://crates.io/crates/handlebars

BOOM

**Comment by SergioBenitez** [comment]
@drusellers Thanks so much for being active about this! (You too, @skariel!) :)

**Comment by drusellers** [comment]
@SergioBenitez ha, however I can help. :)

**Comment by dbrgn** [comment]
:sparkles: 

Any chance for a release with Serde 1.0 support soon-ish?

**Comment by SergioBenitez** [comment]
Yes. 0.3 is nearing its release, which will include Serde 1.0 support.

**Comment by denysvitali** [comment]
When is Rocket 0.3 planned for?

**Comment by SergioBenitez** [comment]
Very soon.

**Comment by denysvitali** [comment]
Thank you for your quick answer. I look forward to it.
If you need help with something let me know, I'll try to help you
